lib/cli.js

```diff
@@ -52,10 +52,10 @@ module.exports = (process) => {
   // this is how to use npm programmatically:
   conf._exit = true
   const updateNotifier = require('../lib/utils/update-notifier.js')
-  npm.load(conf, er => {
+  npm.load(conf, async er => {
     if (er) return errorHandler(er)
 
-    updateNotifier(npm)
+    npm.updateNotification = await updateNotifier(npm)
```

Comment:
hmmm this got me thinking... is there a possibility we could move this to a threaded api (workers/child_process/etc) so the cli never has to wait on it? Simply saying: print it if you had the time, don't bother blocking cli exit in case pacote isn't done fetching the packument yet.

Or is it already the case and I'm missing how something about how it works?

Reply:
Hm.  Yeah, we could do something like:

```
updateNotifier(npm).then(message => { npm.updateNotification = message })
```

and then it'll just get killed when the process exits if it doesn't finish in time.

But it means you won't ever get notified if you're running `npm ls`.  Maybe that's good?  Have to make sure the timer file doesn't get touched before pacote finishes though.

Reply:
So, the `updateTimeout` method would have to be split into two functions, one to check the timer and another to reset it.

Reply:
> Maybe that's good? 

I do think so! but we can def revisit it later 😊 def not a blocker so I ended up merging the PR 😁